test/firebase_auth_mocks_test.dart

```diff
@@ -570,6 +577,51 @@ void main() {
     );
   });
 
+  // Temp variable to hold tUser before provider is linked
+  final _unlinkedTUser = tUser;
+
+  test('User.linkWithProvider works if not already linked', () async {
+    final auth = MockFirebaseAuth(signedIn: true, mockUser: tUser);
+    final user = auth.currentUser;
+    final provider = TwitterAuthProvider();
+    final cred = await user!.linkWithProvider(provider);
+    expect(cred, isNotNull);
+    expect(
+      user.providerData.any((info) => info.providerId == 'twitter.com'),
+      true,
+    );
+  });
+
+  test('User.linkWithProvider throws Exception if already linked', () async {
+    // tUser is still linked to twitter.com from previous test
```

Comment:
It is unsafe to make unit tests have side effects and to expect test orders. It makes it much harder to maintain since you can no longer modify tests independently from one another. The best way would be to combine them into one large test that does things in order.